### PR TITLE
[UTOPIA-482, 483, 486] [Backend] Updated pia-intake findAll endpoint to fetch MPOs ministries PIAs

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- [UTOPIA-482](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-482) - As an MPO, Fetch data for submitted PIA in my ministry 
- [UTOPIA-483](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-483) - Get roles from keycloak
- [UTOPIA-486](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-486) - List all the ministry as an ENUM

After the implementation, the GET `/pia-intake` endpoint will return:
* As a user, retrieve all PIA-intakes I submitted
* As an MPO, retrieve all pia-intakes submitted to my ministry for review

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Tested via swagger and updating roles on the SSO portal 

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [x] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/common/constants/gov-ministries.constant.ts [hunk 1]
```diff
@@ -3,77 +3,102 @@ import { GovMinistriesEnum } from '../enums/gov-ministries.enum';
...
     label: 'Health',
+    code: 'HLTH',
   },
   [GovMinistriesEnum.INDIGENOUS_RELATIONS_AND_RECONCILIATION]: {
     label: 'Indigenous Relations and Reconciliation',
+    code: 'IRR',
   },
   [GovMinistriesEnum.JOBS_ECONOMIC_RECOVERY_AND_INNOVATION]: {
     label: 'Jobs, Economic Recovery and Innovation',
+    code: 'JERI',
```
Comment (Bottle7) [inline src/backend/src/common/constants/gov-ministries.constant.ts:62]: This ministry is now 'Jobs, Economic **Development** and Innovation' with the acronym 'JEDI'

File: src/backend/src/common/constants/gov-ministries.constant.ts [hunk 2]
```diff
@@ -3,77 +3,102 @@ import { GovMinistriesEnum } from '../enums/gov-ministries.enum';
 export const GovMinistries = {
   [GovMinistriesEnum.ADVANCED_EDUCATION_AND_SKILLS_TRAINING]: {
     label: 'Advanced Education and Skills Training',
+    code: 'AEST',
```
Comment (Bottle7) [inline src/backend/src/common/constants/gov-ministries.constant.ts:6]: This ministry is now: Post-Secondary Education and Future Skills, acronym: 'PSEF'(?)

File: src/backend/src/common/constants/gov-ministries.constant.ts [hunk 3]
```diff
@@ -3,77 +3,102 @@ import { GovMinistriesEnum } from '../enums/gov-ministries.enum';
...
     label: 'Jobs, Economic Recovery and Innovation',
+    code: 'JERI',
   },
   [GovMinistriesEnum.LABOUR]: {
     label: 'Labour',
+    code: 'LBR',
   },
   [GovMinistriesEnum.LAND_WATER_AND_RESOURCE_STEWARDSHIP]: {
     label: 'Land, Water and Resource Stewardship',
+    code: 'LWRS',
```
Comment (Bottle7) [inline src/backend/src/common/constants/gov-ministries.constant.ts:70]: This ministry is now: 'Water, Land and Resource Stewardship', acronym: 'WLRS'

File: src/backend/src/common/constants/gov-ministries.constant.ts [hunk 4]
```diff
@@ -3,77 +3,102 @@ import { GovMinistriesEnum } from '../enums/gov-ministries.enum';
...
+    code: 'SDPR',
   },
   [GovMinistriesEnum.TOURISM_ARTS_CULTURE_AND_SPORT]: {
     label: 'Tourism, Arts, Culture and Sport',
+    code: 'TACS',
   },
   [GovMinistriesEnum.TRANSPORTATION_AND_INFRASTRUCTURE]: {
     label: 'Transportation and Infrastructure',
+    code: 'TRAN',
   },
```
Comment (Bottle7) [inline src/backend/src/common/constants/gov-ministries.constant.ts:103]: There are also two new ministries: Ministry of Housing and Ministry of Emergency Management and Climate Readiness.

File: src/backend/src/common/constants/roles.constants.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import { GovMinistriesEnum } from '../enums/gov-ministries.enum';
+import { RolesEnum } from '../enums/roles.enum';
+import { UserTypesEnum } from '../enums/users.enum';
+
+export interface IRoleInfo {
+  user: UserTypesEnum;
+  ministry: GovMinistriesEnum;
+}
+
+export const Roles: Record<RolesEnum, IRoleInfo> = {
```
Comment (Bottle7) [inline src/backend/src/common/constants/roles.constants.ts:10]: The changes mentioned above will, of course, have to be reflected in this file as well.

File: src/backend/src/common/enums/roles.enum.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+export enum RolesEnum {
```
Comment (Bottle7) [inline src/backend/src/common/enums/roles.enum.ts:1]: Changes.

**Comment by kushal-arora-fw** [comment]
@Bottle7  - Valid inputs. However, the changes of/in ministries is a larger topic in terms of impact to the application, and shall not be dealt on a adhoc basis. The changes you mentioned came up in our last week's meet (including Peggy, Mikaela, and JP) and we decided to continue with the existing ministries for now.

Our source of truth for the application will be [this Confluence document](https://apps.itsm.gov.bc.ca/confluence/display/UTOPIA/Current+MPO+and+PCT). Any changes to it - which are assumed to come often, will require JIRA ticket created and then be acted upon. In addition, for the first time, a spike is (or will be) created to understand how changing the ministries impact our application today.

**Review by Bottle7 — APPROVED** [review]

**Comment by binggaofw** [comment]
Looks good to me

**Review by binggaofw — APPROVED** [review]

**Pull request merged**